sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -242,6 +244,12 @@ def wait_for_idle(self, seconds_to_wait=None):
                 self.logger.error(error)
                 raise LivyClientTimeoutException(error)
 
+            if constants.YARN_RESOURCE_LIMIT_MSG in self._logs and \
```

Comment:
unfortunately no. This can only be done by parsing the Yarn diagnostic log. No error code or other API.